utils/concurrentmap/concurrent_map.go

```diff
@@ -1,44 +1,44 @@
-package unittest
+package concurrentmap
 
 import "sync"
 
-// ProtectedMap is a thread-safe map.
-type ProtectedMap[K comparable, V any] struct {
+// ConcurrentMap is a thread-safe map.
+type ConcurrentMap[K comparable, V any] struct {
```

Comment:
similarly, this can be simplified since it's redundant with the namespace
```suggestion
type Map[K comparable, V any] struct {
```